pkg/trace/writer/stats.go

```diff
@@ -95,6 +108,30 @@ func (w *StatsWriter) Run() {
 	}
 }
 
+// FlushSync is a no-op for TraceWriter
+func (w *StatsWriter) FlushSync() error {
+	if !w.syncMode {
+		return log.Error("SyncFlush called on StatsWriter, which is a no-op")
```

Comment:
```suggestion
		return errors.New("Synchronous flushing is disabled. Calling FlushSync will have no effect.")
```
I think returning the error alone is enough. The logging should happen in `(*Agent).FlushSync`